playwright/integration/connect_wallet.test.ts

```diff
@@ -0,0 +1,26 @@
+import { page, baseURL } from '../utils/test_setup'
+import { account } from '../utils/provider_setup'
+
+describe('Connect Wallet', () => {
+  it('Can connect a Wallet', async () => {
+    // GIVEN: At home page
+    await page.goto(baseURL)
+
+    // WHEN: connected a Wallet
+    // Click text=/.*Connect Wallet.*/
+    await page.click('text=/.*Connect Wallet.*/')
```

Comment:
This doesn't look like a specific (HTML) selector but rather (regex?) looking for a specific text string. Unless I misunderstand this piece, is it better to scope down to the selectors so that their presence is validated too?

Reply:
The idea is to do testing from user perspective
`page.click('text=/.*Connect Wallet.*/')` represents
1. User comes to a page
2. User sees text, not a selector, not even necessarily link or button
3. User doesn't care, text looks clickable
4. User clicks

That's what [playwright docs recommend](https://playwright.dev/#version=v1.6.1&path=docs%2Fselectors.md&q=best-practices) and what some other libraries (like react-testing-lib) advocate for

Reply:
Selector presence is for unit (not even integration) testing with libs that like comparing snapshots, like Enzyme
Look up Enzyme vs react-testing-library approaches

Reply:
Does that open up possible collusions/complexity around identical naming conventions for text/labels? Or do you see that as an advantage in terms of enforcing a better user experience for the user? I like the thinking here otherwise, but curious on your thoughts.

Reply:
Although after reading the docs, I wonder if using an actual user-facing attribute like 'aria-label' or 'title' (etc.) would be better. It would thick both the 'selector' and 'user facing attribute/text' boxes.

Reply:
Yes, it does, you are completely right. Trade offs are trade offs
That's why I use `header >> text=Web3` in another click, for specificity

You can even argue, as you are saying, for better user xp. For example we have `Connect Wallet` both in header and in TradeOrders component (when disconnected). But they do the same, so we shouldn't care which user clicks

![dex dev gnosisdev com_ (1)](https://user-images.githubusercontent.com/5121491/99370033-3c00ee00-28ce-11eb-8dac-c17197fa8bd6.png)
![dex dev gnosisdev com_ (2)](https://user-images.githubusercontent.com/5121491/99370044-3e634800-28ce-11eb-8617-45c8e2fb007d.png)

Reply:
But the test will only click the first found item? E.g. I guess we want to be sure to test all possible instance for working properly as expected?

Reply:
Yep
We want to be sure of the whole user flow
So to properly test we would need to do txs
And not rely on rinkeby random address
So setup ganache and deploy contracts locally and who know what else

So yes,and no :wink:

Reply:
Makes sense.